src/vs/workbench/parts/files/browser/views/explorerViewer.ts

```diff
@@ -111,6 +161,55 @@ export class FileDataSource implements IDataSource {
 		}
 	}
 
+	private resolveVirtualDirectories(files: FileStat[]) {
+		if (!this.enableVirtualDirectories) {
+			return;
+		}
+
+		files = files.filter(x => !x.isDirectory);
```

Comment:
I think all of these loops can be condensed to:

```
files.filter(x => !x.isDirectory)
     .foreach((x) => x.isVirtualDirectory = x.isVirtualDirectoryMember = false)
     .foreach((x) => {
		let children = files.filter(child => x !== child && !child.isVirtualDirectoryMember && this.isVirtualDirectoryOf(x.name, child.name));
		
		if(children.length) {
			x.isVirtualDirectory = true;
			x.children = [];
			children.forEach((child) => {
				child.isVirtualDirectoryMember = true;
				x.children.push(child);
			});
		}
	});
```

Assuming this is only called once per folder, I added another check in the filter to make sure a child isn't already a child of another file.  That piece will probably need to be flushed out - I'm not sure how things will work when files match multiple patterns